source/slang/hlsl.meta.slang

```diff
@@ -150,9 +150,36 @@ interface __ITextureShape
     static const int dimensions;
     static const int planeDimensions;
 }
+
+//
 [sealed]
 [builtin]
-interface __ITextureShape1D2D3D : __ITextureShape
+interface __ITextureShapeUnique : __ITextureShape
+{
+}
+__magic_type(TextureShapeSubpassType)
+__intrinsic_type($(kIROp_TextureShapeSubpassType))
+struct __ShapeSubpass : __ITextureShapeUnique
+{
+    static const int flavor = $(SLANG_TEXTURE_SUBPASS);
+    static const int dimensions = 4;
+    static const int planeDimensions = 4;
+}
+__magic_type(TextureType)
+__intrinsic_type($(kIROp_TextureType))
+struct __TextureImplUnique<T, Shape: __ITextureShapeUnique, let isArray:int, let isMS:int, let sampleCount:int, let access:int, let isShadow:int, let isCombined:int, let format:int>
```

Comment:
I don't think we need to separate out __TextureImpl type into different types. Introducing the hierarchy for TextureShaoe should be sufficient.